arangodb-net-standard/IndexApi/IndexApiClient.cs

```diff
@@ -0,0 +1,172 @@
+            using (var response = await _client.GetAsync(uri).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<GetIndexResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Delete an index permanently.
+        /// </summary>
+        /// <param name="indexId">The identifier of the index.</param>
+        /// <returns></returns>
+        public virtual async Task<DeleteIndexResponse> DeleteIndexAsync(string indexId)
+        {
+            string uri = _indexApiPath + "/" +indexId;
+            using (var response = await _client.DeleteAsync(uri).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<DeleteIndexResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+
+        /// <summary>
+        /// Fetch the list of indexes for a collection.
+        /// </summary>
+        /// <param name="query">Query parameters for the request.</param>
+        /// <returns></returns>
+        public virtual async Task<GetAllCollectionIndexesResponse> GetAllCollectionIndexesAsync(GetAllCollectionIndexesQuery query)
+        {
+            string uri = _indexApiPath;
+            if (query == null)
+                throw new System.Exception("query is required");
+            if (string.IsNullOrEmpty(query.CollectionName))
+                throw new System.Exception("Collection name is required");
+
+            uri += '?' + query.ToQueryString();
+            using (var response = await _client.GetAsync(uri).ConfigureAwait(false))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    return DeserializeJsonFromStream<GetAllCollectionIndexesResponse>(stream);
+                }
+                throw await GetApiErrorException(response).ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Creates an index
+        /// </summary>
+        /// <param name="indexType">Type of index to create</param>
+        /// <param name="query">Querystring details</param>
+        /// <param name="body">CreateIndexBody object containing the index details</param>
+        /// <returns></returns>
+        public virtual async Task<IndexResponseBase> PostIndexAsync(IndexType indexType, PostIndexQuery query, PostIndexBody body)
```

Comment:
I noticed that the API docs show the indexType specified as a hash fragment, however I assume that this is not actually required, because the related code here has that part commented out?

So I can see the logic of including it as a method parameter - at least then the API conforms to what the ArangoDB swagger UI docs say, even if we don't actually need to pass it in the URL.

Reply:
@rossmills99, yes, the hash fragment is not required. The API knows what type of index to create by reading the `type` property of the request body. So, I have kept the indexType as a method parameter and all that it does is make sure that the correct value is set in `PostIndexBody.Type`.